### PR TITLE
RSP3a2

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

File: Spec/RestClientPresence.swift [hunk 1]
```diff
@@ -75,9 +75,42 @@ class RestClientPresence: QuickSpec {
...
+                    var realtime = ARTRealtime(options: options)
+                    defer { realtime.close() }
+                    let realtimeChannel = realtime.channels.get("test")
+
+                    realtimeChannel.presence.enterClient("ana", data: "mobile")
+                    realtimeChannel.presence.enterClient("john", data: "desktop")
+                    realtimeChannel.presence.enterClient("john", data: "mobile")
+                    realtimeChannel.presence.enterClient("casey", data: "web")
+                    realtimeChannel.presence.enterClient("casey", data: "mobile")
+                    realtimeChannel.presence.enterClient("john", data: "web")
```
Comment (mattheworiordan) [inline Spec/RestClientPresence.swift]: Why are you entering the same clientId 3 times?

**Comment by mattheworiordan** [comment]
One comment but otherwise good, surprised this didn't pass given it's just REST

File: Spec/RestClientPresence.swift (hunk 1 above)
Comment (ricardopereira) [inline Spec/RestClientPresence.swift]: I was checking if the member was updating the data and I forgot to remove it. Fixed 31f0591.

**Comment by tcard** [comment]
@ricardopereira It's supposed to be REST, but the tests is actually using RealtimePresence.get instead of RestPresence.get.

**Comment by tcard** [comment]
Sorry, scratch that one. Not true, it's correct.

File: Source/ARTPresence+Private.h [hunk 2]
```diff
@@ -11,6 +11,12 @@
 
 #import "ARTPresence.h"
 
+@interface ARTPresenceQuery ()
+
+- (NSMutableArray /* <NSURLQueryItem *> */ *)asQueryItems;
```
Comment (ricardopereira) [inline Source/ARTPresence+Private.h]: `- (__GENERIC(NSMutableArray, NSURLQueryItem *) *)asQueryItems;`?

**Comment by ricardopereira** [comment]
LGTM

File: Source/ARTPresence+Private.h (hunk 2 above)
Comment (tcard) [inline Source/ARTPresence+Private.h]: Fixed at debbfa2.

Comment (ricardopereira) [inline Source/ARTPresence+Private.h]: :+1:

File: Spec/RealtimeClientPresence.swift [hunk 3]
```diff
@@ -426,25 +426,28 @@ class RealtimeClientPresence: QuickSpec {
...
+                        count += 1
+                    }
+
                     waitUntil(timeout: testTimeout) { done in
-                        let partlyDone = AblyTests.splitDone(2, done: done)
-                        channel1.presence.subscribe(.Update) { member in
-                            expect(member.action).to(equal(ARTPresenceAction.Update))
-                            expect(member.clientId).to(equal("john"))
-                            expect(member.data as? NSObject).to(equal("away"))
-                            partlyDone()
```
Comment (tcard) [inline Spec/RealtimeClientPresence.swift]: What was wrong with this exactly?

Comment (ricardopereira) [inline Spec/RealtimeClientPresence.swift]: Honestly, I don't know. I spend so much time with this: the flow is correct, the log doesn't show anything weird, I tried debug the problem but it never happened when I was debugging.

**Pull request merged**